requirements.txt

```diff
@@ -7,11 +7,11 @@
 amqp==5.2.0               # via kombu
 appdirs==1.4.4            # via fs
 attrs==23.2.0             # via jsonschema
-backports-zoneinfo[tzdata]==0.2.1  # via backports-zoneinfo, kombu
+backports.zoneinfo[tzdata]==0.2.1;python_version<"3.9" # via backports-zoneinfo, kombu
```

Comment:
Hi @wirespecter , we usually use `pip-compile` to regenerate the `requirements.txt` file. Normally you would be able to do so by running `reana-dev git-upgrade-requirements -c reana-job-controller`, but this would create a requirements file valid for Python 3.8 and not 3.12.

You can either patch the command by changing this line in `reana/reana/config.py`:
```diff
-PYTHON_DOCKER_IMAGE = "docker.io/library/python:3.8"
+PYTHON_DOCKER_IMAGE = "docker.io/library/python:3.12"
"""Python docker image with the same version as cluster components."""
```

Otherwise you can run a docker container with python 3.12:
```
docker run --rm -it -v /<...path to r-job-controller..>:/code:z docker.io/library/python:3.12 bash -c 'cd /code && pip install --upgrade pip-tools setuptools pip && head requirements.txt && bash'
```

You will find the `pip-compile` command at the top of the `requirements.txt` and it's something like:
```
pip-compile --annotation-style=line --output-file=requirements.txt setup.py
```

Just add `-U` to the end to force pip-compile to upgrade all the packages:
```
pip-compile --annotation-style=line --output-file=requirements.txt setup.py -U
```

Reply:
In any case `backports.zoneinfo` is supposed to make the [`zoneinfo`](https://docs.python.org/3/library/zoneinfo.html) library available for Python<3.9. Now that we are moving to 3.12 that library should not be needed anymore